cmd/abc/abc.go

```diff
@@ -104,9 +107,56 @@ func setLogEnvVars() {
 	}
 }
 
+// checkForUpdates asynchronously checks for updates and prints results to
+// returned buffer.
+// A sync.WaitGroup is returned so main can wait for completion before exiting.
+func checkForUpdates(ctx context.Context) func() {
+	updatesCh := make(chan string, 1)
+
+	go func() {
+		defer close(updatesCh)
+
+		var b bytes.Buffer
+		if err := abcupdater.CheckAppVersion(ctx, &abcupdater.CheckVersionParams{
+			AppID: version.Name,
+			// Version: version.Version,
+			Version: "0.4.0", // intentionally older than cur version to test
+			Writer:  &b,
+		}); err != nil {
+			logging.FromContext(ctx).WarnContext(ctx, "failed to check for new versions",
+				"error", err)
+		}
+
+		select {
+		case updatesCh <- b.String():
+		default:
+		}
+	}()
+
+	return func() {
+		select {
+		case <-ctx.Done():
+			// Context was cancelled
+		case msg := <-updatesCh:
+			logging.FromContext(ctx).WarnContext(ctx, msg)
```

Comment:
I am having an issue where the logger is escaping the output from updater

```
time=2024-03-27T11:44:58.635-07:00 severity=WARNING message="A new version of abc is available! Your current version is 0.4.0. Version 0.7.0 is available at https://github.com/abcxyz/abc.\n\nTo disable notifications for this new version, set ABC_IGNORE_VERSIONS=\"0.7.0\". To disable all version notifications, set ABC_IGNORE_VERSIONS=\"all\".\n"
```

Would we rather just print to stderr? @drevell thoughts?

Reply:
Well, we could be sneaky/hacky and just remove newlines and double quotes from the logged message. `ABC_IGNORE_VERSIONS=\"0.7.0\"` could be just `ABC_IGNORE_VERSIONS=0.7.0`, and so on. This assumes that the version numbers never contain whitespace or shell special characters.

stderr also seems fine to me.

Reply:
I'm not sure we should use the logger for human messages. The logger is for diagnostic messages. This feels like something we want to print to stderr using the cli.